libraries/heft-configuration-loader/src/test/ConfigurationFileLoader.test.ts

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+
+import { ConfigurationFileLoader, PathResolutionMethod, InheritanceType } from '../ConfigurationFileLoader';
```

Comment:
For end users, the error reporting experience is pretty important. It might be useful to have some unit tests for that.

For example, consider a situation like this:

**config/config1.json**
```
{
  "extends": "@rushstack/thing/includes/config2.json",
  "folderPaths": [ "./lib" ]
}
```

**node_modules/@rushstack/thing/includes/config2.json**
```
{
  "extends": "./config3.json"
}
```

**node_modules/@rushstack/thing/includes/config3.json**
```
{
  "filePaths": "value"
}
```

Various interesting errors are possible:
1. `"filePaths"` is supposed to be an array, but they specified a string
2. `"extends"` refers back to an earlier file, creating a circular reference
3. `config3.json` does not exist
4. `config3.json` contains invalid JSON and cannot be parsed at all
5. Everything loads okay, but after all the files are merged together, an error occurs because `"filePaths"` and `"folderPaths"` cannot be specified together. The schema allows either, but not both.

For each of these examples, we can ask:
- Does the error message clearly explain what the problem is?
- Can the end user easily determine which file they need to edit to fix the error?

We don't need to solve everything in this PR, but it would be useful to have some tests to track our progress.